source/kraken/tests/realtime_test.cpp

```diff
@@ -1667,3 +1667,131 @@ BOOST_AUTO_TEST_CASE(ordered_delay_message_test) {
+            ("stop_point:20", "08:20"_t, "08:21"_t, std::numeric_limits<uint16_t>::max(), true, true, 0, 0)
+            ("stop_point:10", "08:30"_t, "08:31"_t, std::numeric_limits<uint16_t>::max(), true, true, 300, 0);
+
+    b.generate_dummy_basis();
+    b.finish();
+    b.data->pt_data->index();
+    b.data->build_raptor();
+    b.data->build_uri();
+
+    auto trip_update_1 = ntest::make_delay_message("vj:1",
+            "20170102",
+            {
+                    std::make_tuple("stop_point:10", "20170102T081000"_pts, "20170102T081100"_pts),
+                    std::make_tuple("stop_point:20", "20170102T082000"_pts, "20170102T082100"_pts),
+                    std::make_tuple("stop_point:10", "20170102T084000"_pts, "20170102T084100"_pts), // Delayed
+            });
+    navitia::handle_realtime("feed", "20170101T1337"_dt, trip_update_1, *b.data);
+
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->lines.size(), 1);
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->routes.size(), 1);
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->vehicle_journeys.size(), 2);
+
+    // Check the original vj
+    auto* vj = b.get<nt::VehicleJourney>("vj:1");
+    auto realtime_vp = vj->rt_validity_pattern()->days;
+    auto base_vp = vj->base_validity_pattern()->days;
+    BOOST_CHECK_MESSAGE(boost::algorithm::ends_with(realtime_vp.to_string(), "1111101"), realtime_vp);
+    BOOST_CHECK_MESSAGE(boost::algorithm::ends_with(base_vp.to_string(), "1111111"), base_vp);
+    BOOST_REQUIRE_EQUAL(vj->stop_time_list.size(), 3);
+
+    // Check the realtime vj
+    vj = b.get<nt::VehicleJourney>("vj:1:modified:0:feed");
+    realtime_vp = vj->rt_validity_pattern()->days;
+    base_vp = vj->base_validity_pattern()->days;
+    BOOST_CHECK_MESSAGE(boost::algorithm::ends_with(realtime_vp.to_string(), "0000010"), realtime_vp);
+    BOOST_CHECK_MESSAGE(boost::algorithm::ends_with(base_vp.to_string(), "0000000"), base_vp);
+    // The realtime vj should have all 3 stop_times but lose the boarding / alighting time on stop_point:10
+    // since it's a lollipop vj and it can't find the base_st
+    BOOST_REQUIRE_EQUAL(vj->stop_time_list.size(), 3);
+    BOOST_REQUIRE_EQUAL(vj->stop_time_list.front().stop_point->uri, "stop_point:10");
+    BOOST_REQUIRE_EQUAL(vj->stop_time_list.front().departure_time, "08:11"_t);
+    BOOST_REQUIRE_EQUAL(vj->stop_time_list.front().boarding_time, "08:11"_t);
+    // SHOULD BE : BOOST_REQUIRE_EQUAL(vj->stop_time_list.front().boarding_time, "08:06"_t);
+    BOOST_REQUIRE_EQUAL(vj->stop_time_list.at(1).stop_point->uri, "stop_point:20");
+    BOOST_REQUIRE_EQUAL(vj->stop_time_list.at(1).departure_time, "08:21"_t);
+    BOOST_REQUIRE_EQUAL(vj->stop_time_list.at(1).boarding_time, "08:21"_t);
+    BOOST_REQUIRE_EQUAL(vj->stop_time_list.at(2).stop_point->uri, "stop_point:10");
+    BOOST_REQUIRE_EQUAL(vj->stop_time_list.at(2).departure_time, "08:41"_t);
+    BOOST_REQUIRE_EQUAL(vj->stop_time_list.at(2).alighting_time, "08:40"_t);
```

Comment:
The arrival_time is 08:40, so the alighting is 08:40. But I meant to check arrival_time on the line above, not departure_time :sweat_smile:. I'll change that.